plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java

```diff
@@ -128,8 +129,6 @@ private T tryConstructor(Constructor<T> constructor, Mapping config, Configurati
                         LOGGER.log(Level.FINER, "The parameter to be set is @Nonnull but is not present; " +
                                                            "setting equal to empty list.");
                         args[i] = Collections.emptyList();
-                    } else if (String.class.isAssignableFrom(t)) {
```

Comment:
this is a behaviour change but as far as I can tell this should never have been merged,
It was a hack to make sonar plugin work better, but sonar plugin should just be fixed.

https://github.com/jenkinsci/configuration-as-code-plugin/pull/950/files#r304891125